### PR TITLE
do not kill thread in case of exception

### PR DESCRIPTION
**Pull request by fceller**
Port of 3.5 change: do not delete thread in case of exception, catch and report the exception instead.

### CODE REVIEW

File: arangod/GeneralServer/IoContext.cpp [hunk 1]
```diff
@@ -38,7 +41,12 @@ IoContext::IoThread::~IoThread() { shutdown(); }
 
 void IoContext::IoThread::run() {
   // run the asio io context
-  _iocontext.io_context.run();
+  try {
+    _iocontext.io_context.run();
+  } catch (std::exception const& ex) {
+    LOG_TOPIC("6794f", WARN, Logger::THREADS)
+      << "caught exception in IO thread '" << ex.what();
```
Comment (jsteemann) [inline arangod/GeneralServer/IoContext.cpp:48]:
```suggestion
      << "caught exception in IO thread: " << ex.what();
```

**Review by jsteemann — APPROVED** [review]

File: arangod/GeneralServer/IoContext.cpp (hunk 1 above)
Comment (fceller) [reply arangod/GeneralServer/IoContext.cpp]: Resolved

**Comment by KVS85** [comment]
http://172.16.10.101:8080/job/arangodb-matrix-pr/13416/

**Comment by KVS85** [comment]
Tests blue.

**Review by KVS85 — APPROVED** [review]
LGTM.

**Pull request merged**